notebook/04_portfolio_trade.ipynb

```diff
@@ -4,378 +4,337 @@
22HY6OjpYvXo16urq8Pnnn2Pr1q349NNPX5q3te9bUlKCf/zjH+jfvz8MDAy4+dauXQsAqKioaPJ9XvW57OpEIhENFNhKR44cgb+/P7S1tdt92YWFhXJjSyiC7to2y7xu2we8Xrvy5ZdfwtPTE7dv38bMmTNbdacmMzMz7N27F3369MGWLVvkbh3fGduqpVryf6clbevOnTuxf/9+pKWlwdfXF3p6ehg/fjxXuOvqjIyMAKDbtMFKSkpYsmQJkpOTMWzYMEyYMAH+/v6dUpwjhJCeigopncjGxgZ5eXnIzc3lO0q7UldXh1AoRFVVVZODncpGxzc2Noa6ujp0dXVRVVUFiUTSaN7CwsJGy9bX14eKigpqa2u5s38vPkaNGtXq3LNnz4aFhQV++OEHVFdX49tvv8XixYvlDjhas26voqSkhJqamkbTWzqIYWu117b7y1/+AqFQiN9++w0DBgx45W26W/u+/v7++PLLL7F48WKkpKSgvr4ejDFs27YNwP/ugNHdxMfHw9bWlu8YCiM7Oxs3b97ErFmzOmT51tbWSExM7JBl80VR2+b28DrtyrVr11BSUoL+/ftj2bJlrb59sYaGBv75z3+CMcbdQh54vW0lEAhe+l6d/X3yIoFAgLlz5+Ly5csoLi5GUFAQGGOYPn06/vWvf3VKhrZISEiAQCCAtbU131HalYmJCfbv34/g4GAkJCTAxcUF//73vxXmBAMhhCgSKqR0Il9fXxgaGmLHjh18R2l3ssuVgoOD5aZXV1fjypUr0NTU5LqLT5gwAQAadT3Nz8/HgwcPGi17+vTpkEqluHXrVqPXNm/eDEtLS0il0lZnVlFRwapVq5Cbm4tvv/0Whw4davJOHa1Zt5cxMTFBZmam3LTs7GxkZGS0OntLtce2EwqFWLNmDYRC4St7o7T2fevq6nDr1i0YGxtj5cqVMDIy4g4gKisrW/ReiigxMRGXL1/usKJAd/THH39AS0uLaz/a24wZM3DmzJkO/f/IB0Vsm9vqddqVx48f47333sMff/yBU6dOQVNTE1OmTEFeXl6r3nvmzJkYOHAgrly5gkuXLnHTW7uttLS05Aoljo6OcpcM8fF90pC+vj6Sk5MBAKqqqhg7dix3h6AXP2td0c6dO+Hr6wsDAwO+o3SIiRMn4v79+/jggw+watUqjBw5EklJSXzHIoSQ7qUjR2AhjX333XdMVVWV3blzh+8oryQbeK6yslJu+rp16xgAFhMTw0178c4QpaWlcneG+Pnnn7l5Hz58yAwNDeXuDJGQkMD8/PyYSCRqNKBhTk4O69OnD7O1tWVnz55lxcXFrKCggP30009MS0urTYP3lpaWMqFQyAQCAZs3b16T87Rm3RhrfrDZFStWMABsx44drKysjD18+JDNmjWLmZmZvXRwwBe3v5+fH1NWVm40/8iRI+UGimWs9dvuxcFmX6W5wWZb876jR49mANg333zD8vLyWEVFBbt69SqztLRkANilS5datF2a+lx2RRKJhA0aNIh5eXkxqVTKdxyFMW7cODZz5swOW35VVRXr168fGz16dJcfcLa7t83t0fa1pl0pKytjrq6u7OTJk9y0a9euMVVVVebt7d3o89DUXXsaCg4OZgDYoEGDuMFMW7utxo8fz4RCIcvIyGC3b99mKioqcndlaa/vk5bM09TnSigUspEjR7J79+6xqqoqlpOTwwIDAxkAtnHjxmbfoyv46aefmJKSErt58ybfUTpFXFwcGzJkCFNVVWXr1q1jVVVVfEcihJDugO7a09nq6uqYv78/MzAwYBEREXzHadKdO3ca3Yngk08+YYyxRtMnTZrE/V1+fj5bvXo1s7GxYaqqqkwoFDI/Pz925cqVRu/x4MEDNnXqVKanp8fdWvHMmTPM19eXW/Z7773HzV9QUMDWrFnDbG1tmaqqKjMyMmLjxo1rdJD9OtauXcsAcHeiaUpL1m3Lli3NbjfGGCsuLmaLFi1iJiYmTFNTkw0fPpxFRkYyd3d3bv5169Y1u/0jIyMbTd+0aRNXzGj4+Pzzz1u97bS1teWW4efn99Lt9uJ7ynbqW/u+eXl57P3332cWFhZMVVWVicVitmDBArZ+/Xpuue7u7q/9uexKysrKmK+vLxOJRCw1NZXvOAqjrKyMqaurs3379nXo+0RHRzMdHR321ltvdcliSndvm9uz7Wtpu7J8+XK5v4+Pj2d5eXmNlvvll18yxhq3kwDY7NmzG63L8OHDuddld9dpzbZKTk5mI0aMYNra2szCwoLt3LlT7vW2fJ+8eA7tdT5XsbGx7P3332dOTk5MS0uLGRoaMi8vL/bLL7906TvhHDt2jKmoqMh9R/YEdXV1bNeuXUxHR4e5uLgoxMk8Qgjp4gIFjHXTAQi6sKqqKrz55psICQnBTz/9hHnz5vEdiRDSwVJSUvDmm28iNzcXFy9exIABA/iOpDCOHz+OmTNnIisrq9Ggp+3t6tWrmDJlCgYPHozDhw9zg1ISQhQXYwxff/01Pv30U6xYsQLbt2/nOxIvHj9+jKVLl+Ly5ctYtGgRtm7d2uRA1IQQQl5pA42RwgMNDQ2cOnUKy5cvx4IFCzBt2jSFuGUrIaT1pFIptm7dCjc3N2hqaiIqKoqKKK0UHBwMLy+vDi+iAMDo0aNx69YtPH78GC4uLjh27FiHvychpOM8evQIo0aNwueff47t27f32CIK8OdNDy5cuIBDhw7h+PHj6Nu3L4KCgviORQghCokKKTxRVlbGli1bEBoaiqSkJNjb22P9+vUoLS3lOxohpJ1cvnwZgwYNwscff4yVK1fi5s2bsLCw4DuWQmGM4dy5c5g0aVKnvaerqyvu3buHqVOnYtasWRgzZgxiY2M77f0JIW0nkUgQGBgIFxcXFBQU4M6dO1ixYgXfsbqEmTNn4v79+xgxYgSmTZuG+fPno6ioiO9YhBCiUKiQwrMRI0YgJiYGn332GXbt2oU+ffrgiy++aHSrSfJqAoHglY/AwEC
```

Comment:
We want to link to this [notebook](https://github.com/rapidsai/gQuant/blob/master/notebook/05_customize_nodes.ipynb) for customized GPU kernel in Numba.

Reply:
Done as follows:

In case that there is no direct dataframe method for a particular logic, a Numba GPU kernel can be used to implement it. Some examples about customized GPU kernels in Numba can be found [here](https://github.com/rapidsai/gQuant/blob/master/notebook/05_customize_nodes.ipynb).